src/main/java/com/marklogic/client/dataloader/impl/DefaultSchemasDataLoader.java

```diff
@@ -0,0 +1,47 @@
+package com.marklogic.client.dataloader.impl;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.dataloader.SchemasDataFinder;
+import com.marklogic.client.dataloader.SchemasDataLoader;
+import com.marklogic.client.document.TextDocumentManager;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.helper.LoggingObject;
+import com.marklogic.client.io.FileHandle;
+import com.marklogic.client.io.Format;
+
+public class DefaultSchemasDataLoader extends LoggingObject implements SchemasDataLoader {
+	
+	@Override
+	public Set<File> loadSchemasData(File baseDir, SchemasDataFinder schemasDataFinder, DatabaseClient client) {
+		XMLDocumentManager xmlDocMgr = client.newXMLDocumentManager();
+		TextDocumentManager textDocMgr = client.newTextDocumentManager();
+		
+		List<File> schemasData = schemasDataFinder.findSchemasData(baseDir);
+
+        Set<File> loadedSchemasData = new HashSet<>();
+
+        for (File f : schemasData) {
+        	String extension = getExtensionNameFromFile(f);
+        	FileHandle handle = new FileHandle(f);
+        	if (extension.equals("tde") || extension.equals("xsd")) {
+        		xmlDocMgr.write(f.getName(), handle.withFormat(Format.XML));
```

Comment:
This is just a Client API question - is it possible to use one Manager here and thus a DocumentWriteSet? Not saying to do it here - I doubt the number of schemas would ever be such that a write set really helps - I'm just curious about the Client API.